gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -421,6 +417,18 @@ public int read(ByteBuffer byteBuffer) throws IOException {
     }
 
     long effectivePosition = positionInGrpcStream + bytesToSkipBeforeReading;
+
+    /* resIterator is null on the first read (position = 0) or when a seek is performed (and when
+      there are exceptions). So if footerBuffer is null and we are trying to read into footer
+      region, we may just cache the footer
+    */
+    if ((resIterator == null)
+        && (footerBuffer == null)
+        && (effectivePosition >= footerStartOffsetInBytes)) {
```

Comment:
we are still missing out on overlap. How about 
```
effectivePosition + buffer.size() >= footerStartOffsetInBytes
```

This would avoid any redundant data transmission if the file footer offset is before configured footer .

Reply:
I thought about this. But I also read up a little on parquet file format. The length of the footer is stored at the end of the footer. So the first read typically happens at the very end. I don't think optimization will be helpful in reading larger footer.